xenserver/tests/test_views.py

```diff
@@ -0,0 +1,101 @@
+"""
+Some quick and dirty tests for a very small subset of the code.
+"""
+
+from django.core.urlresolvers import reverse
+import pytest
+
+from xenserver.models import (
+    Addresses, AddressPool, Project, Template, XenServer, XenVM, Zone)
+from xenserver import tasks
+from xenserver.tests.matchers import ANY_VALUE
+
+
+@pytest.mark.django_db
+class TestProvision(object):
+    def setup_template(self, name, cores=1, memory=2048, diskspace=10240,
+                       iso="installer.iso"):
+        return Template.objects.create(
+            name=name, cores=cores, memory=memory, diskspace=diskspace,
+            iso=iso)
+
+    def setup_project(self, name):
+        return Project.objects.create(name=name)
+
+    def setup_server(self, hostname):
+        zone = Zone.objects.create(name="foozone")
+        AddressPool.objects.create(
+            subnet="10.0.0.0/24", gateway="10.1.2.3", zone=zone, version=4)
+        mem = 64*1024*1204*1024
+        return XenServer.objects.create(
+            hostname=hostname, username="u", password="p", zone=zone,
+            memory=mem, mem_free=mem)
+
+    def test_provision_simple(self, monkeypatch, settings, admin_client):
+        """
+        We can create a new VM using mostly default values.
+        """
+        createvm_calls = []
+        monkeypatch.setattr(tasks, 'getSession', lambda *a, **kw: "session")
+        monkeypatch.setattr(
+            tasks, '_create_vm', lambda *a: createvm_calls.append(a))
+        settings.CELERY_ALWAYS_EAGER = True
```

Comment:
Maybe a comment about this `monkeypatch` and `CELERY_ALWAYS_EAGER` stuff.. I don't really know what's going on there..